### PR TITLE
Create todo feature

### PR DESCRIPTION
**Pull request by coder4097**
## Description
Created an endpoint to add a new `todo` item.

URI: POST /todos
req.body: { title: … }
res status: 201
res: { id: , title: }

### Database schema changes

New model Todo added.
Schema:
{
title: { type: String, required: true }
},
{
collection: 'todo',
timestamps: {
createdAt: 'createdAt',
updatedAt: 'updatedAt',
},
}

### Automated test cases added
- should create a new todo if valid string is passed as "title" in body, it should return 201
- should give validation error if empty string is passed as "title" in body, status 400
- should give a validation error when title is not a string, status 400

### Manual test cases run

1. URI: POST /todos
req.body: { "title" : "title after review" }
res.status: 201
res: {
"id": "6114b0a5831b7c11f0306e64",
"title": "title after review"
}

URI: POST /todos
req.body: { "title" : "" }
res.status: 400
res: {
"message": "Validation failed. Please modify the request and try again.",
"failures": [
{
"field": "title",
"message": "Please specify the valid title"
}
]
}

### CODE REVIEW

File: src/controllers/todo-controller.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import { TodoItem } from "@models";
+import {
+  AppContext,
+  Errors,
+  ExtendedRequest,
+  ValidationFailure,
+} from "@typings";
+import { createTodoValidator } from "@validators";
+
+export class TodoController extends BaseController {
```
Comment (coder1033) [inline src/controllers/todo-controller.ts:13]: Replace Todo with TodoItem. `TodoItemController`
Make this change everywhere.

File: src/storage/mongoose/todo.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { model, Schema } from "mongoose";
+
+const todoSchema: Schema = new Schema(
+  {
+    title: { type: String, required: true },
+    isActive: { type: Boolean, default: true },
+  },
+  {
+    collection: "todo",
```
Comment (coder1033) [inline src/storage/mongoose/todo.ts:9]: `todos`

File: test/mocks/in-memory-mongo-store.ts [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
 
 export class InMemoryMongoStore extends Mongoose.MongoStore {
   constructor() {
     super();
   }
 
   public async connect(): Promise<M> {
-    const mongod = new MongoMemoryServer();
-    const uri = await mongod.getConnectionString();
+    const uri: string = config.get("mongoDb.uri");
```
Comment (coder1033) [inline test/mocks/in-memory-mongo-store.ts:12]: Use `mongod.getUri()`. We should use MongoMemoryServer instead of our database for testing.

**Comment by coder1033** [comment]
No manual testing?

File: src/controllers/todo-controller.ts (hunk 1 above)
Comment (coder4097) [reply src/controllers/todo-controller.ts:13]: updated

File: src/storage/mongoose/todo.ts (hunk 2 above)
Comment (coder4097) [reply src/storage/mongoose/todo.ts:9]: done

File: test/mocks/in-memory-mongo-store.ts (hunk 3 above)
Comment (coder4097) [reply test/mocks/in-memory-mongo-store.ts:12]: updated

File: src/controllers/todo-controller.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import { TodoItem } from "@models";
+import {
+  AppContext,
+  Errors,
+  ExtendedRequest,
+  ValidationFailure,
+} from "@typings";
+import { createTodoItemValidator } from "@validators";
+
+export class TodoItemController extends BaseController {
```
Comment (coder1033) [inline src/controllers/todo-controller.ts:13]: File name - `todo-item-controller.ts`
Make this similar change to every file name.

Comment (coder4097) [reply src/controllers/todo-controller.ts]: updated

**Review by coder1033 — APPROVED** [review]

**Pull request merged**